src/marine/copernicus_l3swh2ioda.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
+
+#
+# (C) Copyright 2021 UCAR
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+
+import sys
+import argparse
+import netCDF4 as nc
+import numpy as np
+from datetime import datetime, timedelta
+import pytz
+import os
+from pathlib import Path
+
+IODA_CONV_PATH = Path(__file__).parent/"../lib/pyiodaconv"
+if not IODA_CONV_PATH.is_dir():
+    IODA_CONV_PATH = Path(__file__).parent/'..'/'lib-python'
+sys.path.append(str(IODA_CONV_PATH.resolve()))
+
+import ioda_conv_engines as iconv
+from collections import defaultdict, OrderedDict
+from orddicts import DefaultOrderedDict
+
+locationKeyList = [
+    ("latitude", "float"),
+    ("longitude", "float"),
+    ("datetime", "string"),
+]
+
+obsvars = {
+    'swh': 'swh',
+}
+
+AttrData = {
+    'converter': os.path.basename(__file__),
+    'nvars': np.int32(len(obsvars)),
+}
+
+DimDict = {
+}
+
+VarDims = {
+    'swh': ['nlocs'],
+}
+
+
+class copernicus(object):
+    def __init__(self, filename, factor):
+        # Create a simple data structure for Copernicus L3 swh near-realtime product obtained from
+        # ftp://nrt.cmems-du.eu/Core/WAVE_GLO_WAV_L3_SWH_NRT_OBSERVATIONS_014_001/
+        ncd = nc.Dataset(filename, 'r')
+        self.lons = np.squeeze(ncd.variables['longitude'][:]).ravel()
```

Comment:
No need for _squeezing_ and your arrays are already 1D, so no need for _ravel_ . That's true for all the lines below making use of these 2 methods.

Reply:
Thx for point this out, @guillaumevernieres ! I have removed these methods.